src/ui/application-window.vala

```diff
@@ -283,6 +313,26 @@ namespace Peek.Ui {
         }
       }
     }
+    
+    
+    
+
+    [GtkCallback]
+    private void on_gif_button_clicked (Button source) {
+       select_format("gif");
+       }
+    [GtkCallback]
+    private void on_apng_button_clicked (Button source) {
+       select_format("apng");
+    	}
+    [GtkCallback]
+    private void on_webm_button_clicked (Button source) {
+        select_format("webm");
+       }
+    [GtkCallback]
+    private void on_mp4_button_clicked (Button source) {
+       select_format("mp4");
+    	}
```

Comment:
This code duplication screams for some kind of unification where the format is a parameter, but I don't know if that's possible when using an XML UI definition.

Reply:
Yeah, if you find a more elegant solution let me know, we could use radiobuttons and group them that is the only option I can think of at the moment.